replay-test/run.ts

```diff
@@ -0,0 +1,287 @@
+// Test runner for recording node scripts and replaying the resulting recordings.
+
+// Using require() here to workaround typescript errors.
+import * as fs from "fs";
+import * as path from "path";
+import { spawnSync, SpawnSyncReturns } from "child_process";
+import { TestManifest, NodeTestIgnoreList } from "./manifest";
+import { listAllRecordings, uploadRecording } from "@recordreplay/recordings-cli";
+import ProtocolClient from "./client";
+
+const Usage = `
+Usage: node run <options>
+Options:
+  --node <path>: Specify the path to the node binary to use to record.
+  --api-key <key>: Specify the API key to use for recording and replaying.
+  --run-suite: Run the default replay test suite.
+  --run-random-tests <count>: Run random tests from node's regular test suite.
+  --run-pattern <pattern>: Run all tests matching a pattern.
+  --server <address>: Set server to connect to (default wss://dispatch.replay.io).
+`;
+
+function bailout(message) {
+  console.log(message);
+  process.exit(1);
+}
+
+if (process.argv.length == 2) {
+  bailout(Usage);
+}
+
+let gNodePath: string;
+let gAPIKey: string;
+let gRunSuite: boolean;
+let gRunRandomTests: number;
+let gRunPattern: string;
+let gDispatchAddress = "wss://dispatch.replay.io";
+
+for (let i = 2; i < process.argv.length; i++) {
+  switch (process.argv[i]) {
+    case "--node":
+      gNodePath = path.resolve(process.cwd(), process.argv[++i]);
+      break;
+    case "--api-key":
+      gAPIKey = process.argv[++i];
+      break;
+    case "--run-suite":
+      gRunSuite = true;
+      break;
+    case "--run-random-tests":
+      gRunRandomTests = +process.argv[++i];
+      break;
+    case "--run-pattern":
+      gRunPattern = process.argv[++i];
+      break;
+    case "--server":
+      gDispatchAddress = process.argv[++i];
+      break;
+    default:
+      bailout(Usage);
+  }
+}
+
+if (!gNodePath) {
+  bailout("Node path not specified");
+}
+
+if (!gAPIKey) {
+  bailout("API key not specified");
+}
+
+if (!gRunSuite && !gRunRandomTests && !gRunPattern) {
+  bailout("No tests specified");
+}
+
+const gRecordingDirectory = path.join(__dirname, `recordings-${(Math.random() * 1e9) | 0}`);
+
+let gNumFailures = 0;
+
+async function main() {
+  fs.mkdirSync(gRecordingDirectory);
```

Comment:
nit: we could use [the `fs/promises` node API](https://nodejs.org/api/fs.html#promises-api) and `await fs.mkdir(gRecordingDirectory)`, etc. in this function to make things a bit more consistent. Not a blocker for me though.